### DISCUSSION TITLE
vscode debugging unbound breakpoints

### DISCUSSION THREAD

**Discussion by JustinBoileau** [requarks/wiki#7310] [Help / Questions] [answered]
I have installed a development environment on a windows 11 PC , using vscode & docker desktop, as per the instructions on the web site.

When I attempt to set any breakpoints in vscode e.g server/modules/rendering/markdown-core/renderer.js, they appear as 'unbound breakpoints'.

I am new to js development and unsure how to resolve the issue. Any help is appreciated.

**Comment by Mexator** [comment, marked as answer]
In package.json there is scripts block:
```json
"scripts": {
    "start": "node server",
    "dev": "NODE_OPTIONS=--openssl-legacy-provider node dev",
```

Find "dev" script and add `--inspect-brk=9222` option to it:

```json
"scripts": {
    "start": "node server",
    "dev": "NODE_OPTIONS=--openssl-legacy-provider node --inspect-brk=9222 dev",
```

Run your app as per "[Getting started](https://docs.requarks.io/dev#running-the-project)" instructions.

After executing `yarn run dev` open the "Run and Debug" tab run "Attach to port".

**Reply by JustinBoileau** [reply]
Thank you works great
